test/nextStepCliText.test.ts

```diff
@@ -209,12 +259,97 @@ describe(`Next steps cli text`, () => {
+      Promise.resolve(true) /** doesYarnrcYmlFileExist */,
+      () =>
+        Promise.resolve(
+          some({
+            dependencies: {
+              [targetProject]: '0.0.0',
+              [anotherTargetProject]: '0.0.0',
+            },
+          })
+        ) /** getPackageJsonContents */,
+      () => Promise.resolve(false) /** isYarnBerryUsingNodeModulesLinker */
+    );
+
+    expect(virtualLoggerLogSpy).toHaveBeenCalled();
+    expect(logger.lastMessage()).toMatch('Error: Detected you are using yarn 2+');
+
+    done();
+  });
+
+  it(`should output yarn commands if yarn 2+ is detected`, async done => {
+    const workingDirectory = 'destination';
+    const targetProject = 'widgets';
+    const anotherTargetProject = 'anotherTargetProject';
+    const yetAnotherTargetProject = 'yetAnotherTargetProject';
+    const logger = getVirtualLoggerInstance();
+    const virtualLoggerLogSpy = spyOn(logger, 'log').and.callThrough();
+
+    await maybeOutputNextStepsText(
+      '/tmp/build' /* maybeDestinationDirectoryToAddDependencyOn */,
+      [
+        { targetProjectName: targetProject, targetProjectAbsolutePath: targetProject },
+        { targetProjectName: anotherTargetProject, targetProjectAbsolutePath: anotherTargetProject },
+        { targetProjectName: yetAnotherTargetProject, targetProjectAbsolutePath: yetAnotherTargetProject },
+      ],
+      true /* outputPostCommandMessages */,
+      logger,
+      workingDirectory,
+      Promise.resolve(false) /** doesPackageLockFileExist */,
+      Promise.resolve(true) /** doesYarnLockFileExist */,
+      Promise.resolve(true) /** doesYarnrcYmlFileExist */,
+      () =>
+        Promise.resolve(
+          some({
+            dependencies: {
+              [targetProject]: '0.0.0',
+            },
+            devDependencies: {
+              [anotherTargetProject]: '0.0.0',
+            },
+          })
+        ) /** getPackageJsonContents */,
+      () => Promise.resolve(true) /** isYarnBerryUsingNodeModulesLinker */
+    );
+
+    const outputMatchRegexText =
+      `^\n\nSet up target projects as local dependencies with yarn using:\n` +
+      `\tyarn add widgets@file:/tmp/build/widgets\n` +
+      `\tyarn add -D anotherTargetProject@file:/tmp/build/anotherTargetProject yetAnotherTargetProject@file:/tmp/build/yetAnotherTargetProject\n` +
+      `\tyarn install\n\n.*`;
+    const outputMatch = new RegExp(outputMatchRegexText);
+
+    expect(virtualLoggerLogSpy).toHaveBeenCalled();
+    expect(logger.lastMessage()).toMatch(outputMatch);
+
+    done();
+  });
+
   it(`should not output anything if project setup as a local dependency`, async done => {
     const workingDirectory = 'destination';
     const targetProject = 'widgets';
     const anotherTargetProject = 'anotherTargetProject';
     const logger = getVirtualLoggerInstance();
-    const virtualLoggerLogSpy = spyOn(logger, 'log');
+    const virtualLoggerLogSpy = spyOn(logger, 'log').and.callThrough();
```

Comment:
Not strictly necessary in these tests, but I spent 45 minutes debugging why `lastMessage()` wasn't working when I copied-and-pasted them.  I figured it was better to be consistent.